backends/dpdk/dpdkProgram.cpp

```diff
@@ -134,6 +95,66 @@ const IR::DpdkAsmProgram *ConvertToDpdkProgram::create(IR::P4Program *prog) {
                     });
     statements.push_back(s);
 
+    IR::IndexedVector<IR::DpdkHeaderType> headerType;
+    for (auto kv : structure.header_types) {
+        LOG3("add header type " << kv.second);
+        auto h = kv.second;
+        auto ht = new IR::DpdkHeaderType(h->srcInfo, h->name, h->annotations,
+                                         h->fields);
+        headerType.push_back(ht);
+    }
+    IR::IndexedVector<IR::DpdkStructType> structType;
+
+    /* Insert the metadata structure updated with tmp variables created during parser conversion */
+    auto new_objs = new IR::Vector<IR::Node>;
+    for (auto obj : prog->objects) {
+        if (auto s = obj->to<IR::Type_Struct>()) {
+            if (s->name.name == info->local_metadata_type) {
+                auto *annotations = new IR::Annotations(
```

Comment:
why don't you factor this code into a separate function that returns a IndexedVector?

Reply:
I will do that.